File: java/gov/nasa/gsfc/gmsec/api/mist/FieldSpecification.java
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/**
 * @file FieldSpecification.java
 */

package gov.nasa.gsfc.gmsec.api.mist;


/**
 * The FieldSpecification class represents a single field definition from a message
 * schema.
 *
 * @see MessageSpecification
 * @see Specification
 */
public class FieldSpecification
{
	private String m_name;
	private String m_type;
	private String m_mode;
	private String m_class;
	private String m_value;
	private String m_desc;


	private FieldSpecification()
	{
	}


	private FieldSpecification(String name, String type, String mode, String clazz, String value, String desc)
	{
		m_name  = name;
		m_type  = type;
		m_mode  = mode;
		m_class = clazz;
		m_value = value;
		m_desc  = desc;
	}


	/**
	 * Returns the name for the Field Specification.
	 *
	 * @return The name associated with the Field Specification.
	 */
	public String getName()
	{
		return m_name;
	}


	/**
	 * Returns the type (e.g. STRING, I16, etc.) for the Field Specification.
	 *
	 * @return The field type associated with the Field Specification.
	 */
	public String getType()
	{
		return m_type;
	}


	/**
	 * Returns the mode (e.g. REQUIRED, OPTIONAL, etc.) for the Field Specification.
	 *
	 * @return The mode associated with the Field Specification.
	 */
	public String getMode()
	{
		return m_mode;
	}


	/**
	 * Returns the field class type for the Field Specification.
	 *
	 * @return The class type associated with the Field Specification.
	 */
	public String getClassification()
	{
		return m_class;
	}


	/**
	 * Returns the default value, if any, for the Field Specification. If a
	 * value is not available, then an empty string is returned.
	 *
	 * @return The default value associated with the Field Specification.
	 */
	public String getValue()
	{
		return m_value;
	}


	/**
	 * Returns the description for the Field Specification.
	 *
	 * @return The description associated with the Field Specification.
	 */
	public String getDescription()
	{
		return m_desc;
	}
}

File: framework/src/gmsec4/mist/FieldSpecification.cpp
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

#include <gmsec4/mist/FieldSpecification.h>

#include <gmsec4/internal/mist/InternalFieldSpecification.h>


using namespace gmsec::api;
using namespace gmsec::api::mist;
using namespace gmsec::api::mist::internal;


FieldSpecification::FieldSpecification(const char* name, const char* type, const char* mode, const char* clazz, const char* value, const char* desc)
	: m_internal(new InternalFieldSpecification(name, type, mode, clazz, value, desc))
{
}


FieldSpecification::~FieldSpecification()
{
	delete m_internal;
}


const char* FieldSpecification::getName() const
{
	return m_internal->getName();
}


const char* FieldSpecification::getType() const
{
	return m_internal->getType();
}


const char* FieldSpecification::getMode() const
{
	return m_internal->getMode();
}


const char* FieldSpecification::getClassification() const
{
	return m_internal->getClassification();
}


const char* FieldSpecification::getValue() const
{
	return m_internal->getValue();
}


const char* FieldSpecification::getDescription() const
{
	return m_internal->getDescription();
}
